src/multi_select.coffee

```diff
@@ -98,12 +107,26 @@ Ember.Widgets.MultiSelectComponent = Ember.Widgets.SelectComponent.extend
     @set 'values', Ember.A [] unless @get('values')
 
   deletePressed: (event) ->
-    if event.target.selectionStart == 0
+    if event.target.selectionStart == 0 and event.target.selectionEnd == 0
       @removeSelectItem(@get('selections.lastObject'))
 
   removeSelectItem: (item) ->
     @get('selections').removeObject item
 
+  focusIn: (event) ->
+    @addBorderMultiSelectContainer()
+
+  focusOut: (event) ->
+    @removeBorderMultiSelectContainer()
+
+  keyDown: (event) ->
+    @_super(event)
+    activeElement = $(document.activeElement)[0]
+    if @$()[0].contains(activeElement) or @$()[0]==activeElement
```

Comment:
nit: use `is` instead of == and use proper spacing

nit: local variable with meaningful name for `@$()[0]`